app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CartsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :current_user_cart
+
+  def show
+    @books = @cart.books
+    @total_price = @books.sum(&:price)
+  end
+  def add_item
+    @book = Book.find_by!(id: cart_params[:book_id])
+    Order.create(cart: @cart, book: @book)
+    flash.now[:notice] = "The book - #{@book.title.capitalize} is added to your cart."
+    respond_to do |format|
+      format.turbo_stream
+      format.html { render body: "" }
+    end
+  end
+
+  def remove_item
+    @book = Book.find_by!(id: cart_params[:book_id])
+    order = Order.find_by!(cart_id: @cart.id, book_id: @book.id)
+    order.destroy!
```

Comment:
Can we use a single query to destroy order?
```suggestion
    @book = Book.find_by!(id: cart_params[:book_id])
    Order.where(cart_id: @cart.id, book_id: @book.id).destroy
```

Reply:
Can we use `delete_all` then? We can make fewer SQL calls that way.

We don't really need `destroy!`, personally I prefer handling error here rather than using exception.

Reply:
> Can we use delete_all then? We can make fewer SQL calls that way.

Hmm, I try to run `Order.find_by!(cart_id: @cart.id, book_id: @book.id).destroy` (**method_1**) and `Order.where(cart_id: @cart.id, book_id: @book.id).destroy_all` (**method_2**) in rails console and application. I saw both of them generate 2 SQL queries. Additionally, the **method_1** use `limit` to get one record, I think it will be faster than `where` - to get all records. Why do you think if we use **method_2**, we can make fewer SQL calls?

Reply:
I mean `delete_all`, not `destroy_all`. It should use a single query, something like this:

```sql
DELETE FROM orders WHERE card_id = ? AND book_id = ?
```

Reply:
Oops! That's my fault. I got it. Thank you.